packages/components/src/internal/components/search/FilterFacetedSelector.tsx

```diff
@@ -26,21 +28,30 @@ export const FilterFacetedSelector: FC<Props> = memo(props => {
     const [fieldDistinctValues, setFieldDistinctValues] = useState<string[]>(undefined);
     const [error, setError] = useState<string>(undefined);
     const [searchStr, setSearchStr] = useState<string>(undefined);
+    const [allShown, setAllShown] = useState<boolean>(true);
 
     useEffect(() => {
+        setDistinctValues(true);
+    }, [fieldKey]); // on fieldKey change, reload selection values
+
+    const setDistinctValues = useCallback((checkAllShown: boolean, searchStr?: string) => {
+        const filterArray = searchStr ? [Filter.create(fieldKey, searchStr, Filter.Types.CONTAINS)].concat(fieldFilters) : fieldFilters;
```

Comment:
I don't think fieldFilters should be used here. should concat selectDistinctOptions?.filterArray instead. selectDistinctOptions?.filterArray would contain other field filters but exclude current field fitlers.

Reply:
Done. (so many different filters...)